cmd/abapEnvironmentBuild.go

```diff
@@ -0,0 +1,270 @@
+package cmd
+
+import (
+	"encoding/json"
+	"time"
+
+	abapbuild "github.com/SAP/jenkins-library/pkg/abap/build"
+	"github.com/SAP/jenkins-library/pkg/abaputils"
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+type abapEnvironmentBuildUtils interface {
+	command.ExecRunner
+	abaputils.Communication
+	abapbuild.Publish
+	abapbuild.HTTPSendLoader
+	getMaxRuntime() time.Duration
+	getPollingIntervall() time.Duration
+}
+
+type abapEnvironmentBuildUtilsBundle struct {
+	*command.Command
+	*piperhttp.Client
+	*abaputils.AbapUtils
+	maxRuntime       time.Duration
+	pollingIntervall time.Duration
+}
+
+func (aEBUB *abapEnvironmentBuildUtilsBundle) getMaxRuntime() time.Duration {
+	return aEBUB.maxRuntime
+}
+
+func (aEBUB *abapEnvironmentBuildUtilsBundle) getPollingIntervall() time.Duration {
+	return aEBUB.pollingIntervall
+}
+
+func (aEBUB *abapEnvironmentBuildUtilsBundle) PersistReportsAndLinks(stepName, workspace string, reports, links []piperutils.Path) {
+	abapbuild.PersistReportsAndLinks(stepName, workspace, reports, links)
+}
+
+func newAbapEnvironmentBuildUtils(maxRuntime time.Duration, pollingIntervall time.Duration) abapEnvironmentBuildUtils {
+	utils := abapEnvironmentBuildUtilsBundle{
+		Command: &command.Command{},
+		Client:  &piperhttp.Client{},
+		AbapUtils: &abaputils.AbapUtils{
+			Exec: &command.Command{},
+		},
+		maxRuntime:       maxRuntime * time.Minute,
+		pollingIntervall: pollingIntervall * time.Second,
+	}
+	// Reroute command output to logging framework
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
+func abapEnvironmentBuild(config abapEnvironmentBuildOptions, telemetryData *telemetry.CustomData, cpe *abapEnvironmentBuildCommonPipelineEnvironment) {
+	utils := newAbapEnvironmentBuildUtils(time.Duration(config.MaxRuntimeInMinutes), time.Duration(config.PollingIntervallInSeconds))
+	err := runAbapEnvironmentBuild(&config, telemetryData, utils, cpe)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runAbapEnvironmentBuild(config *abapEnvironmentBuildOptions, telemetryData *telemetry.CustomData, utils abapEnvironmentBuildUtils, cpe *abapEnvironmentBuildCommonPipelineEnvironment) error {
+	values, err := generateValues(config)
+	if err != nil {
+		return errors.Wrap(err, "Generating the values from config failed")
+	}
+	conn := new(abapbuild.Connector)
+	if err := initConnection(conn, config, utils); err != nil {
+		return errors.Wrap(err, "Connector initialization for communication with the ABAP system failed")
+	}
+	finalValues, err := runBuild(conn, config, utils, values)
+	if err != nil {
+		return errors.Wrap(err, "Error during execution of build framework")
+	}
+	cpe.build.values = finalValues
```

Comment:
can this be chosen freely? maybe cpe.abap.buildValues ?

Reply:
Yes, I would just need to edit this in the yaml file

Reply:
Done